orderBook.js

```diff
@@ -0,0 +1,43 @@
+const reconcileOrder = (existingBook, incomingOrder) =>
+  existingBook.length
+    ? checkAllTypesMatch(existingBook, incomingOrder)
+    : addToBook(existingBook, incomingOrder)
+
+const checkAllTypesMatch = (existing, incoming) =>
+  existing.every((obj) => obj.type === incoming.type)
+    ? addToBook(existing, incoming)
+    : checkPricesMatch(existing, incoming)
+
+const checkPricesMatch = (existing, incoming) => {
+  for (let i = 0; i < existing.length; i++) {
+    if (existing[i].price === incoming.price) {
+      return reconcileQuantities(existing, incoming)
+    }
+  }
+  return addToBook(existing, incoming)
+}
+
+const reconcileQuantities = (existing, incoming) => {
+  for (let i = 0; i < existing.length; i++) {
+    if (existing[i].quantity === incoming.quantity) {
+      existing.splice(i, 1)
+      return existing
+    }
+    if (existing[i].quantity > incoming.quantity) {
+      existing[i].quantity -= incoming.quantity
+      return existing
+    }
+    if (existing[i].quantity < incoming.quantity) {
+      incoming.quantity -= existing[i].quantity
+      existing.splice(i, 1)
+      return checkAllTypesMatch(existing, incoming)
+    }
+  }
+}
```

Comment:
This function has a for loop and it is called inside a for loop which is called from inside an every. A triple nested loop is still gross even when broken into different functions.